### PR TITLE
Add hit-pressure detection.

### PR DESCRIPTION
**Pull request by J-Wass**
Design over here: https://docs.google.com/document/d/1yz_Us7yVLfy8fmeQ5sR9dMarb4VUr2X9l8Sy5jzcpzo/edit?usp=sharing

Implements hit.pressure which ranges from 100 to 0. Pressure of 0 means a hit was made with little to no defender interference. Pressure of 100 means a defender is nearly on top of the hit. 

Implementation wise, pressure detection takes the current defender's position and velocity and estimates how many frames (in 0.1 frame increments) it would take before the defender blocks a hit.

### CODE REVIEW

File: carball/tests/stats/hit_test.py [hunk 1]
```diff
@@ -122,3 +122,26 @@ def test(analysis: AnalysisManager, answer):
...
+             100, 100, 100, 100, 0, 0, 0, 0, 0, 0, 0, 94, 0, 100, 97, 100, 100,
+             0, 100, 0, 0, 0, 0, 0, 81, 100, 97, 100, 81, 100, 0, 100, 100, 0,
+             100, 100, 100, 100, 91, 0, 0, 0, 100, 0, 96, 100, 100, 99, 0, 0, 0,
+             100, 0, 0, 0, 0, 0, 100, 0, 99, 100, 98, 0, 94, 0, 100, 0, 100,
+             100, 0, 100, 100, 100, 0, 100, 0, 100, 67, 0, 91, 100, 100, 100,
+             0, 0, 0, 0, 88, 0, 100, 0, 100, 0, 0]
+            for x in range(len(hits)):
+                assert(hits[x].pressure == expected_pressures[x])
+
+                run_analysis_test_on_replay(test, replay_list=get_raw_replays()["OCE_RLCS_7_CARS"],
```
Comment (dtracers) [inline carball/tests/stats/hit_test.py]: This test doesn't actually run because the call to run it is inside the for loop

Comment (J-Wass) [reply carball/tests/stats/hit_test.py]: Thanks, it should be running now.

File: carball/tests/stats/hit_test.py [hunk 2]
```diff
@@ -122,3 +122,26 @@ def test(analysis: AnalysisManager, answer):
...
+             100, 100, 100, 100, 0, 0, 0, 0, 0, 0, 0, 94, 0, 100, 97, 100, 100,
+             0, 100, 0, 0, 0, 0, 0, 81, 100, 97, 100, 81, 100, 0, 100, 100, 0,
+             100, 100, 100, 100, 91, 0, 0, 0, 100, 0, 96, 100, 100, 99, 0, 0, 0,
+             100, 0, 0, 0, 0, 0, 100, 0, 99, 100, 98, 0, 94, 0, 100, 0, 100,
+             100, 0, 100, 100, 100, 0, 100, 0, 100, 67, 0, 91, 100, 100, 100,
+             0, 0, 0, 0, 88, 0, 100, 0, 100, 0, 0]
+            for x in range(len(hits)):
+                assert(hits[x].pressure == expected_pressures[x])
+
+        run_analysis_test_on_replay(test, replay_list=get_raw_replays()["OCE_RLCS_7_CARS"],
```
Comment (dtracers) [inline carball/tests/stats/hit_test.py]: Can you have it commented that it can't use the cache because it is running intensive commands

Comment (J-Wass) [reply carball/tests/stats/hit_test.py]: Done.

File: carball/analysis/events/hit_pressure/pressure_analysis.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            ball_position = [ball_row.pos_x,ball_row.pos_y,ball_row.pos_z]
+            player_position = [player_row.pos_x,player_row.pos_y,player_row.pos_z]
+            player_velocity = [player_row.vel_x/30,player_row.vel_y/30,player_row.vel_z/30]
+
+            original_distance = self.get_distance(ball_position, player_position)
+            # Only consider defenders within 2000uu of hit.
+            if original_distance < 2000:
+                # Check the next 10 frames to see if the opposing player would have hit the ball.
+                for i in range(100):
+                    frame = i/10
```
Comment (Sciguymjm) [inline carball/analysis/events/hit_pressure/pressure_analysis.py]: What's the purpose of this? Is there a reason to use `range(100)`?

Comment (J-Wass) [reply carball/analysis/events/hit_pressure/pressure_analysis.py]: Arbitrarily give defenders 10 frames to be within blocking range of the ball. Mostly via experimentation I arrived at that number. There may a more scientific way to determine which frame distance is best to stop checking at, however.

**Pull request merged**